libsolidity/formal/Z3Interface.cpp

```diff
@@ -107,6 +113,33 @@ pair<CheckResult, vector<string>> Z3Interface::check(vector<Expression> const& _
 	return make_pair(result, values);
 }
 
+Expression* Z3Interface::maximize(Expression const& _expressionToMaximize)
+{
+	z3::optimize::handle handler = m_optimizer.maximize(toZ3Expr(_expressionToMaximize));
```

Comment:
Is it also possible to do that in plain smtlib2?

Reply:
This code is a little different in the latest commit. If I'm not mistaken, `maximize` and `minimize` are not part of the standard `smtlib2`, but `z3` and `MathSAT` do accept them in the extended language.